sky/resources.py

```diff
@@ -920,12 +920,6 @@ def _try_validate_ports(self) -> None:
         """
         if self.ports is None:
             return
-        if skypilot_config.get_nested(('aws', 'security_group_name'),
```

Comment:
For a customized security group, it is possible that the security group does not the required ports opened and the user does not have enough permission to modify the security group to add those rules.
Also, checking the ports that are already open for a security group is a bit tricky, as it is possible that some of the rules open up some ports but only expose them to a specific range of IP.
https://github.com/skypilot-org/skypilot/blob/05aafb8b5970acb353eac24c4ee7ef0c6a97e118/sky/provision/aws/instance.py#L718-L729

Tagging @cblmemo for anything to chime in here.

In the long run, we may have to handle those, but it might be fine for now to print a WARNING / hint here to let a user know that the security group is specified and it is not guaranteed that the ports will be correctly opened on AWS.

```python
if self.cloud is None or instance(self.cloud, clouds.AWS):
    security_group_name = skypilot_config.get_nested(('aws', 'security_group_name'), None)
    if security_group_name is not None:
            with ux_utils.print_exception_no_traceback():
                logger.warning(
                    f'Ports {self.ports} and security group name are specified: {security_group_name}. '
                    'It is not guaranteed that the ports will be opened if the specified security group is not correctly set up. '
                    'Please try to specify `ports` only and leave out `aws.security_group_name` in `~/.sky/config.yaml` to allow SkyPilot to automatically create and configure the security group.')
```

Reply:
The warning looks good to me. Random idea: could also consider remove the port specification for load balancer if we use security groups as resoruces.

Reply:
I've added the suggested warning.